flink-queryable-state/flink-queryable-state-runtime/src/main/java/org/apache/flink/queryablestate/client/proxy/KvStateClientProxyHandler.java

```diff
@@ -133,12 +131,11 @@ private void executeActionAsync(
 			operationFuture.whenCompleteAsync(
 					(t, throwable) -> {
 						if (throwable != null) {
-							if (throwable instanceof CancellationException) {
-								result.completeExceptionally(throwable);
-							} else if (throwable.getCause() instanceof UnknownKvStateIdException ||
+							if (
+									throwable.getCause() instanceof UnknownKvStateIdException ||
```

Comment:
Do i understand it correctly that in the case of a wrong KvStateId a UnknownKvStateLocationException is thrown?

Reply:
Then, in which cases is an UnknownKvStateIdException thrown? That actually sounds more appropriate.

Reply:
No, an `UnknownKvStateIdException` is thrown in this case. See `KvStateServerHandler`.

Reply:
The `UnknownKvStateLocationException` is thrown in the case we do not have a server for a particular `keyGroup`.

Reply:
Then i don't understand how this can fix the issue.

In case of an `UnknownKvStateIdException` the client still retries fetching the state.

Reply:
The problem was the, now removed, `UnknownKvStateLocation`. This is the one returned by the JobManager. Then, based on that, you go to the appropriate TaskManager.